src/cmd/flux-job.c

```diff
@@ -603,6 +626,175 @@ int cmd_id (optparse_t *p, int argc, char **argv)
+        return -1;
+    return 0;
+}
+
+void eventlog_continuation (flux_future_t *f, void *arg)
+{
+    struct eventlog_ctx *ctx = arg;
+    double timestamp;
+    char name[FLUX_KVS_MAX_EVENT_NAME + 1];
+    char context[FLUX_KVS_MAX_EVENT_CONTEXT + 1];
+    char buf[64];
+    const char *s;
+    const char *event;
+
+    if (flux_job_eventlog_lookup_get (f, &s) < 0)
+        log_err_exit ("flux_job_eventlog_lookup_get");
+
+    if (flux_kvs_eventlog_append (ctx->log, s) < 0)
+        log_err_exit ("flux_kvs_eventlog_append");
+
+    while ((event = flux_kvs_eventlog_next (ctx->log))) {
+        if (flux_kvs_event_decode (event, &timestamp, name, sizeof (name),
+                                   context, sizeof (context)) < 0)
+            log_err_exit ("flux_kvs_event_decode");
+
+        if (eventlog_timestr (timestamp, buf, sizeof (buf)) < 0)
+            log_msg_exit ("error converting timestamp to ISO 8601");
+
+        printf ("%s %s%s%s\n", buf, name, *context ? " " : "", context);
+    }
+    fflush (stdout);
+    flux_future_destroy (f);
+}
+
+int cmd_eventlog (optparse_t *p, int argc, char **argv)
+{
+    flux_t *h;
+    int optindex = optparse_option_index (p);
+    flux_jobid_t id;
+    flux_future_t *f;
+    char *endptr;
+    struct eventlog_ctx ctx = {0};
+
+    if (!(h = flux_open (NULL, 0)))
+        log_err_exit ("flux_open");
+
+    if (argc - optindex != 1) {
+        optparse_print_usage (p);
+        exit (1);
+    }
+    id = strtoull (argv[optindex++], &endptr, 10);
+    if (*endptr != '\0')
+        log_err_exit ("id requires an unsigned integer argument");
+
+    if (!(ctx.log = flux_kvs_eventlog_create()))
+        log_err_exit ("flux_kvs_eventlog_create");
+    ctx.p = p;
+
+    if (!(f = flux_job_eventlog_lookup (h, id)))
+        log_err_exit ("flux_job_eventlog_lookup");
+    if (flux_future_then (f, -1., eventlog_continuation, &ctx) < 0)
+        log_err_exit ("flux_future_then");
+    if (flux_reactor_run (flux_get_reactor (h), 0) < 0)
+        log_err_exit ("flux_reactor_run");
+
+    flux_kvs_eventlog_destroy (ctx.log);
+    flux_close (h);
+    return (0);
+}
+
+struct wait_event_ctx {
+    optparse_t *p;
+    const char *wait_event;
+    double timeout;
+};
+
+void wait_event_continuation (flux_future_t *f, void *arg)
+{
+    struct wait_event_ctx *ctx = arg;
+    char name[FLUX_KVS_MAX_EVENT_NAME + 1];
+    const char *s;
+
+    if (flux_rpc_get (f, NULL) < 0) {
+        if (errno == ETIMEDOUT) {
+            fprintf (stderr, "wait-event timeout on event '%s'\n",
+                     ctx->wait_event);
+            flux_future_destroy (f);
+            exit (1);
+        } else if (errno == ENODATA) {
+            flux_future_destroy (f);
+            return;
+        }
+        /* else fallthrough and have `flux_job_event_watch_get'
+         * handle error */
+    }
+
+    if (flux_job_event_watch_get (f, &s) < 0)
+        log_err_exit ("flux_job_event_watch_get");
```

Comment:
Suggestion, on at least ENOENT, return a more readable error to the user, e.g. "job %lu not found" instead of `flux_job_event_watch_get: No such file or directory`